### PR TITLE
Chore!: cleanup unnecessary tokens

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -3941,10 +3938,12 @@ def _parse_window_spec(self) -> t.Dict[str, t.Optional[str | exp.Expression]]:
 
         return {
             "value": (
-                self._match_set((TokenType.UNBOUNDED, TokenType.CURRENT_ROW)) and self._prev.text
+                (self._match_text_seq("UNBOUNDED") and "UNBOUNDED")
+                or (self._match_text_seq("CURRENT", "ROW") and "CURRENT ROW")
+                or self._parse_bitwise()
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:3943]: We have another `self._parse_bitwise()` right below this, is this needed?

File: sqlglot/parser.py [hunk 2]
```diff
@@ -607,7 +589,6 @@ class Parser(metaclass=_Parser):
         "LOCK": lambda self: self._parse_locking(),
         "LOCKING": lambda self: self._parse_locking(),
         "LOG": lambda self: self._parse_log(no=self._prev.text.upper() == "NO"),
-        "MATERIALIZED": lambda self: self.expression(exp.MaterializedProperty),
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:610]: Why was this removed?

File: sqlglot/parser.py [hunk 3]
```diff
@@ -4125,8 +4124,10 @@ def _parse_wrapped(self, parse_method: t.Callable, optional: bool = False) -> t.
             self._match_r_paren()
         return parse_result
 
-    def _parse_select_or_expression(self) -> t.Optional[exp.Expression]:
-        return self._parse_select() or self._parse_set_operations(self._parse_expression())
+    def _parse_select_or_expression(self, alias: bool = False) -> t.Optional[exp.Expression]:
```
[the remark this answers is not in the document]
Comment (tobymao) [reply sqlglot/parser.py:4127]: because only in struct and pivots do you do struct(1 as y)

File: sqlglot/parser.py (hunk 2 above)
Comment (tobymao) [reply sqlglot/parser.py:610]: was never used, no tests for this

Comment (tobymao) [reply sqlglot/parser.py:610]: i'll add it back and a test

Comment (GeorgeSittas) [reply sqlglot/parser.py:610]: We do have one:

```sql
CREATE MATERIALIZED VIEW x.y.z AS SELECT a FROM b
```

File: sqlglot/parser.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/parser.py:3922]: thanks

File: sqlglot/parser.py (hunk 2 above)
Comment (tobymao) [reply sqlglot/parser.py:610]: it didn't fail when i removed it

File: sqlglot/parser.py [hunk 4]
```diff
@@ -1262,15 +1227,28 @@ def _parse_create(self) -> t.Optional[exp.Expression]:
         )
 
     def _parse_property_before(self) -> t.Optional[exp.Expression]:
+        # only used for teradata currently
         self._match(TokenType.COMMA)
 
         # parsers look to _prev for no/dual/default, so need to consume first
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:1233]: Should we update or get rid of this comment?

**Comment by GeorgeSittas** [comment]
Looks good, I'll do another pass in a bit. Great refactor!

File: sqlglot/parser.py [hunk 5]
```diff
@@ -1428,46 +1396,48 @@ def _parse_checksum(self) -> exp.Expression:
...
-            return self.expression(exp.DataBlocksizeProperty, default=True)
-        elif self._match_texts(("MIN", "MINIMUM")):
-            self._match_text_seq("DATABLOCKSIZE")
-            return self.expression(exp.DataBlocksizeProperty, min=True)
-        elif self._match_texts(("MAX", "MAXIMUM")):
-            self._match_text_seq("DATABLOCKSIZE")
-            return self.expression(exp.DataBlocksizeProperty, min=False)
-
-        self._match_text_seq("DATABLOCKSIZE")
+    def _parse_datablocksize(self, default=None, minimum=None, maximum=None) -> exp.Expression:
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:1427]: Type hints

File: sqlglot/parser.py [hunk 6]
```diff
@@ -1338,7 +1316,6 @@ def _parse_properties(self, before=None) -> t.Optional[exp.Expression]:
         return None
 
     def _parse_fallback(self, no=False) -> exp.Expression:
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:1318]: Type hint

File: sqlglot/parser.py [hunk 7]
```diff
@@ -1397,19 +1374,10 @@ def _parse_withjournaltable(self) -> exp.Expression:
         return self.expression(exp.WithJournalTableProperty, this=self._parse_table_parts())
 
     def _parse_log(self, no=False) -> exp.Expression:
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:1376]: Type hint

File: sqlglot/parser.py [hunk 8]
```diff
@@ -1428,46 +1396,48 @@ def _parse_checksum(self) -> exp.Expression:
...
+        )
+
     def _parse_freespace(self) -> exp.Expression:
-        self._match_text_seq("FREESPACE")
         self._match(TokenType.EQ)
         return self.expression(
             exp.FreespaceProperty, this=self._parse_number(), percent=self._match(TokenType.PERCENT)
         )
 
     def _parse_mergeblockratio(self, no=False, default=False) -> exp.Expression:
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:1414]: Type hints

**Review by GeorgeSittas — APPROVED** [review]

File: tests/test_parser.py [hunk 9]
```diff
@@ -498,3 +498,8 @@ def test_pivot_columns(self):
                 expr = parse_one(query, read=dialect)
                 columns = expr.args["from"].this.args["pivots"][0].args["columns"]
                 self.assertEqual(expected_columns, [col.sql(dialect=dialect) for col in columns])
+
+    def test_parse_properties(self):
+        self.assertEqual(
+            parse_one("create materialized table x").sql(), "CREATE MATERIALIZED TABLE x"
+        )
```
[the remark this answers is not in the document]
Comment (tobymao) [reply tests/test_parser.py:505]: no because it's not an identity check

[the remark this answers is not in the document]
Comment (GeorgeSittas) [reply tests/test_parser.py:505]: Ah you wanted to ensure the lowercase keywords don't affect the logic?
